pkg/kv/dist_sender.go

```diff
@@ -1125,6 +1130,9 @@ func (ds *DistSender) sendPartialBatch(
 			if err := evictToken.EvictAndReplace(ctx, replacements...); err != nil {
 				return response{pErr: roachpb.NewError(err)}
 			}
+			if ba.IsUnsplittable() {
+				return response{pErr: pErr}
```

Comment:
huh, I think I messed this one up and it actually needs to move into `divideAndSendBatchToRanges`, right after the check if the batch fits within a (new) single range.